homework/list.js

```diff
@@ -1,79 +1,95 @@
 'use strict';
 
-(function() {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+var List = function() {
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper')
+}
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
+var ListNote = new List() ;
+
+List.prototype.start=function() {
+    var f = this.updateList.bind(this);
+    
+    this.fetchList()
+        
+        /*.then(function(data){
+        this.updateList(data);
+        }.bind(this))*/
+
+        .then(this.updateList.bind(this))
+        //.then(f)
+        .then(this.drawList.bind(this))
+        .then(this.preloadFirstNote.bind(this));
+        //.then(this.preloadFirstNote().bind(this));
+        //It is wrong because ()
+    
     window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+        this.onNoteOpen(event);
+    }.bind(this));
+   
+}
 
-  function onNoteOpen(event) {
+List.prototype.onNoteOpen = function(event) {
```

Comment:
If a function body is wrapped by a conditional statement only with one branch, it's better to put the `if...else` in the caller, rather than the callee.